js/private/js_library.bzl

```diff
@@ -82,6 +82,19 @@ _ATTRS = {
         The transitive npm dependencies, transitive sources & runfiles of targets in the `deps` attribute are added to the
         runfiles of this target. They should appear in the '*.runfiles' area of any executable which is output by or has a
         runtime dependency on this target.
+
+        Npm dependencies in `deps` are propagated to the transitive npm dependencies of downstream `npm_package` targets when
+        they are linked.
+        """,
+        providers = [JsInfo],
+    ),
+    "dev_deps": attr.label_list(
+        doc = """Development dependencies of this target.
```

Comment:
"development" means compile-time but not runtime? Is there a better term for that? :/

Reply:
npm docs `"devDependencies": Packages that are only needed for local development and testing."` tho that doesn't translate over perfectly to rules_js. That is the idea, however, when it comes propagating deps through the linker.

https://docs.npmjs.com/specifying-dependencies-and-devdependencies-in-a-package-json-file